### PR TITLE
[CSLD-77] CLI-docs chapter

### PR DESCRIPTION
**Pull request by denisshevchenko**

### CODE REVIEW

File: src/addr-convert/Main.hs [hunk 1]
```diff
@@ -1,15 +1,66 @@
...
+optionsParser = do
+    address <- optional $ textOption $
+           short   'a'
+        <> long    "address"
+        <> help    "Address to convert. It must be in base64(url) format."
+        <> metavar "STRING"
+    return AddrConvertOptions{..}
+
+getAddrConvertOptions :: IO AddrConvertOptions
+getAddrConvertOptions = execParser programInfo >>= return
```
Comment (Martoon-00) [inline src/addr-convert/Main.hs]: Lil

File: src/smart-generator/GenOptions.hs [hunk 2]
```diff
@@ -41,58 +44,85 @@ optionsParser = do
...
     goInitBalance <- option auto $
         long "init-money" <>
-        help "How many coins node has in the beginning"
+        help "How many coins node has in the beginning."
     goInitTps <- option auto $
         short   't' <>
         long    "tps" <>
         metavar "DOUBLE" <>
-        help    "TPS (transactions per second)"
+        help    "TPS (transactions per second),"
```
Comment (Martoon-00) [inline src/smart-generator/GenOptions.hs]: Why comma?

Comment (denisshevchenko) [inline src/smart-generator/GenOptions.hs]: ops, it should be dot..

File: src/Pos/CLI.hs [hunk 3]
```diff
@@ -142,13 +141,15 @@ sysStartParser :: Opt.Parser Timestamp
...
     Opt.metavar "TIMESTAMP" <>
+    Opt.help    helpMsg <>
     defaultValue
   where
     -- In development mode, this parameter is mandatory.
     -- In production mode, it is optional, and its default value is populated
     -- from `staticSysStart`, which gets it from the config file.
     defaultValue =
         if isDevelopment then mempty else Opt.value staticSysStart
+    helpMsg = "Time when system started. Mandatory in development mode. Format - seconds since Unix-epoch."
```
Comment (volhovM) [inline src/Pos/CLI.hs]: System start time. Compulsory.

File: src/Pos/CLI.hs [hunk 4]
```diff
@@ -287,10 +287,9 @@ externalNetworkAddressOption na =
          <> Opt.showDefault
          <> maybe mempty Opt.value na
   where
-    helpMsg = "Ip and port of external address. "
-        <> "Please mind that you need to specify actual accessible "
-        <> "ip of host, at which node is run,"
-        <> " otherwise work of CSL is not guaranteed. "
+    helpMsg = "IP and port of external address. "
+        <> "Please make sure these IP and port (at which node is running) are actually accessible "
```
Comment (volhovM) [inline src/Pos/CLI.hs]: s/at/on/
s/actually//g (actually has a different meaning, i can't find something to exchange it with tho)

File: src/Pos/CLI.hs [hunk 5]
```diff
@@ -302,19 +301,13 @@ listenNetworkAddressOption na =
...
   where
-    helpMsg = "Ip and port on which to bind and listen."
-
-sysStartOption :: Opt.Parser Timestamp
-sysStartOption = Opt.option (Timestamp . sec <$> Opt.auto) $ 
-    Opt.long    "system-start" <>
-    Opt.metavar "TIMESTAMP" <>
-    Opt.value   staticSysStart
+    helpMsg = "IP and port on which to bind and listen. Please make sure these IP "
+        <> "and port are actually accessible, otherwise proper work of CSL isn't guaranteed."
```
Comment (volhovM) [inline src/Pos/CLI.hs]: Same as above. Maybe you can remove duplication here.

File: src/dht-keygen/Main.hs [hunk 6]
```diff
@@ -27,10 +32,32 @@ unHashId (HashId bs) = bs
...
+    versionOption = infoOption
+        ("cardano-dht-keygen-" <> showVersion version)
+        (long "version" <> help "Show version.")
 
 main :: IO ()
-main = mapM_ processArg =<< getArgs
+main = do
+    KeyGenOptions{..} <- getKeyGenOptions
+    key <- generateKey $ BS.pack nonce
+    putStrLn . fromMaybe "Invalid nonce length, it must contain 14 characters." . fmap encodeUrl $ key
```
Comment (volhovM) [inline src/dht-keygen/Main.hs]: length must _be_ 14 chars

File: src/keygen/KeygenOptions.hs [hunk 7]
```diff
@@ -43,22 +50,22 @@ data FakeAvvmOptions = FakeAvvmOptions
...
         long    "genesis-file" <>
         metavar "FILE" <>
         value   "genesis.bin" <>
-        help    "File to dump binary shared genesis data"
+        help    "File to dump binary shared genesis data."
     koRearrangeMask <- optional $ strOption $
         long    "rearrange-mask" <>
         metavar "PATTERN" <>
-        help    "Secret keyfiles to rearrange"
+        help    "Secret keyfiles to re-arrange."
```
Comment (volhovM) [inline src/keygen/KeygenOptions.hs]: It's "rearrange" after all:
https://answers.yahoo.com/question/index?qid=20121202112942AAzC1VU

File: src/node/NodeOptions.hs [hunk 8]
```diff
@@ -79,120 +83,143 @@ argsParser = do
...
-        help "If we DB already exist, discard its contents \
-             \and create a new one from scratch"
+        help "If node's database already exists, discard its contents \
+             \and create a new one from scratch."
     devSpendingGenesisI <- if isDevelopment
         then (optional $ option auto $
                   long    "spending-genesis" <>
                   metavar "INT" <>
-                  help    "Use genesis spending #i")
+                  help    "Index of using secret key in genesis.")
```
Comment (volhovM) [inline src/node/NodeOptions.hs]: Maybe "used genesis secret key index" or something? Here and below (VSS).

File: update_cli_docs.sh [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+echo "Cardano SL CLI docs updating"
+
+readonly CARDANO_DOCS_REPO="${HOME}"/cardanodocs
+readonly DOCUMENTER=cardano-cli-docs
+readonly RAW_CHAPTER_NAME="${DOCUMENTER}".md
+readonly PATH_TO_REAL_CHAPTER="${CARDANO_DOCS_REPO}"/_docs/technical/2017-01-15-cli-options.md
+readonly PATH_TO_DIR_WITH_EXECUTABLES=$(stack path --local-install-root)/bin
+
+echo "**** 1. Get raw Markdown chapter ****"
+stack exec --nix -- "${DOCUMENTER}" --bin-dir "${PATH_TO_DIR_WITH_EXECUTABLES}"
```
Comment (volhovM) [inline update_cli_docs.sh]: We don't force `--nix` anywhere. It ought to be set in `~/.stack/config.yaml` or something.

File: src/analyzer/AnalyzerOptions.hs [hunk 10]
```diff
@@ -1,16 +1,23 @@
+{-# LANGUAGE QuasiQuotes #-}
 
--- | Command line options of pos-node.
```
Comment (gromakovsky) [inline src/analyzer/AnalyzerOptions.hs]: ы

File: src/genupdate/Main.hs [hunk 11]
```diff
@@ -2,34 +2,77 @@ module Main
...
+import           Paths_cardano_sl             (version)
+import qualified Codec.Archive.Tar            as Tar
+import qualified Control.Foldl                as Fold
+import           Crypto.Hash                 (Digest, SHA512, hashlazy)
+import qualified Data.ByteString.Lazy        as BSL
+import           Data.List                   ((\\))
+import           Filesystem.Path             (filename)
+import           Turtle                      hiding (f, s, toText)
+import           Turtle.Prelude              (stat)
+import           Universum                   hiding (FilePath, fold)
```
Comment (gromakovsky) [inline src/genupdate/Main.hs]: Do you use `stylish-haskell`? 😕

File: src/node/NodeOptions.hs [hunk 12]
```diff
@@ -79,120 +83,143 @@ argsParser = do
         long    "db-path" <>
         metavar "FILEPATH" <>
         value   "node-db" <>
-        help    "Path to the node database"
+        help    "Path to directory with node’s database (called Global State). \
```
Comment (gromakovsky) [inline src/node/NodeOptions.hs]: No, it's not part to Global State. Global State in only one of DBs. This directory contains all DBs used by node.

File: update_cli_docs.sh (hunk 9 above)
Comment (gromakovsky) [inline update_cli_docs.sh]: I think this script in only for build system, so shouldn't matter as long as build system has nix.

**Pull request merged**